specs/agents/tracing-distributed-tracing.md

```diff
@@ -1,27 +1,88 @@
-In order to make sure we are fully aligned, all agents are implementing the
-specification described in [this commit](https://github.com/w3c/trace-context-binary/blob/571cafae56360d99c1f233e7df7d0009b44201fe/spec/20-binary-format.md).
+The `traceparent` header is composed of four parts:
 
-#### Tracestate
+ * `version`
+ * `trace-id`
+ * `parent-id`
+ * `trace-flags`
+
+Example:
+
+```
+traceparent: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01
+             () (______________________________) (______________) ()
+             v                 v                        v         v
+           Version         Trace-Id                 Span-Id     Flags
+```
+
+
+#### Version
+
+The `version` is 1 byte (2 hexadecimal digits) representing an 8-bit unsigned
+integer. Currently, the `version` will always be `00`.
+
+#### Trace ID
+
+A Trace ID is globally unique, and consists of 128 random bits (like a UUID).
+Its string representation is 32 hexadecimal digits.  This is the ID for the
+whole distributed trace and stays constant throughout a given trace.
+
+#### Transaction/Span ID, and Parent ID
+
+Each transaction object will store the global `trace_id`. If the transaction
+is started without an incoming `traceparent` header, then the `trace_id`
+should be generated.
+
+Each transaction and span object will have an `id`. This is generated for each
+transaction and span, and is 64 random bits (with a string representation of
+16 hexadecimal digits).
+
+Each transaction and span object will have a `parent_id`, except for the
+very first transaction in the distributed trace. The `parent_id` will be the
+`id` of the parent transaction/span. For new transactions with an incoming
+`traceparent` header, the `span-id` piece of the `traceparent` should be
+used as the `parent_id`.
+
+In addition to the above rules, spans will also have a `transaction_id`,
+which is the `id` of the current transaction. While not necessary for
+distributed tracing, this inclusion allows for simpler and more performant UI
+queries.
+
+Error objects will also include the `trace_id` (optional), an `id` (which in
+the case of errors is 128 bits, encoded as 32 hexadecimal digits), a
+`transaction_id`, and a `parent_id` (which is the `id` of the transaction or
+span that caused the error).
+
+
+#### Flags
+
+The W3C traceparent header specifies 8 bits for flags. Currently, only a single
+flag (`sampled`) is defined, with the rest reserved for later use. These flags
+are recommendations given by the by the caller rather than strict rules to
+follow.
+
+##### Sampled
+
+The `sampled` flag is the least significant bit (right-most) and denotes that
+the caller may have recorded trace data. If this flag is unset (`0` in the
+least significant bit), the agent should not sample the transaction. If this
+flag is set (`1` in the least significant bit), the agent should sample the
+transaction. See the [sampling](tracing-sampling.md) specification for more
+details.
```

Comment:
Should we add note like `the agent may ignore this flag if sampling a transaction would conflict with another config option, e.g. rate limit`

Reply:
@jahtalab Could you please clarify under which circumstances "the agent may ignore this flag"?
I assumed that sampling decision is made only by the agent recording the root transaction of the trace (i.e., the transaction without incoming `traceparent`) and the downstream agents have to obey that sampling decision by using `sampled` flag of incoming `traceparent`, am I missing something?

Reply:
@SergeyKleyman , that is the way agents currently handle the flag, my point was mainly in regards to rate limiting, so that agent devs know they can ignore this flag if it violates the rate limit config. We can also discuss this separately from this PR and add it later.

Reply:
@jahtalab Thank you for the clarification - I just wanted to make sure my implementation covers all the use cases.